mcopy: add page (#21998)
tldr-pages/tldr: pages/common/mcopy.md

# mcopy

> Copy MSDOS files to/from Unix.

> > Part of the `mtools` package. See also: `mdir`, `mdel`, `mmove`.

Reviewer: ```suggestion
> Part of the `mtools` package.
> See also: `mdir`, `mdel`, `mmove`.
> Note: Use forward slash (`/`) instead of backlash (`\`) to refer to DOS subdirectories.
```

> More information: <https://manned.org/mcopy>.

- Copy a file from Linux to an MS-DOS disk or image:

> `mcopy {{path/to/file}} {{a:}}`

Reviewer: ```suggestion
`mcopy {{path/to/source_file}} {{A}}:{{/path/to/target_file}}`
```

Some DOS filesystems can support subdirectories.

Write drive letters in uppercase, see https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#paths-1 (might be updated to support DOS).

- Copy a file from an MS-DOS disk to the current Linux directory:

> `mcopy {{a:file.txt}} .`

Reviewer: ```suggestion
`mcopy {{A}}:{{/path/to/source_file}} .`
```

- Copy all files from an MS-DOS drive to a Linux directory:

> `mcopy {{a:}} {{path/to/directory}}`

Reviewer: ```suggestion
`mcopy {{A}}: {{path/to/target_directory}}`
```

> - Copy a file to a disk image using the `-i` flag instead of a drive letter:

Reviewer: ```suggestion
- Copy a file into a DOS disk [i]mage file instead of a drive letter (`::` represents the root directory of the DOS disk):
```

Use short option mnemonics. https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#short-option-mnemonics

> `mcopy -i {{path/to/image.img}} {{path/to/file}} {{::}}`

Reviewer: ```suggestion
`mcopy -i {{path/to/image.img}} {{path/to/source_file}} ::/{{path/to/target_file}}`
```

- Copy a file from a disk image to the current directory:

> `mcopy -i {{path/to/image.img}} {{::file.txt}} .`

Reviewer: ```suggestion
`mcopy -i {{path/to/image.img}} ::/{{path/to/source_file}} .`
```

> - Copy a directory recursively to an MS-DOS disk:

Reviewer: ```suggestion
- Copy a directory recur[s]ively to an MS-DOS disk:
```

> `mcopy -s {{path/to/directory}} {{a:}}`

Reviewer: ```suggestion
`mcopy -s {{path/to/source_directory}} {{A:/path/to/target_directory}}`
```

> - Copy a file and preserve attributes:

Reviewer: ```suggestion
- Copy a file and [p]reserve attributes:
```

> `mcopy -p {{path/to/file}} {{a:}}`

Reviewer: ```suggestion
`mcopy -p {{path/to/source_file}} {{A:}}`
```

# mcopy

> Copy MSDOS files to/from Unix.
> Part of the `mtools` package.

> > See also: `mdir`, `mdel`, `mmove`.

Reviewer 2: These pages do not exist yet, so they can be removed for the meantime.

```suggestion

```

> Note: Use forward slash (`/`) instead of backlash (`\`) to refer to DOS subdirectories.

> > More information: <https://manned.org/mcopy>.

Reviewer 2: ```suggestion
> More information: <https://gnu.org/software/mtools/manual/mtools.html#mcopy>.
```

- Copy a file from Linux to an MS-DOS disk or image:

> `mcopy {{path/to/source_file}} {{A}}:{{/path/to/target_file}}`

Reviewer: ```suggestion
`mcopy {{path/to/source_file}} {{A}}:/{{path/to/target_file}}`
```

- Copy a file from an MS-DOS disk to the current Linux directory:

`mcopy {{A}}:{{/path/to/source_file}} .`

- Copy all files from an MS-DOS drive to a Linux directory:

`mcopy {{A}}: {{path/to/target_directory}}`

- Copy a file into a DOS disk [i]mage file instead of a drive letter (`::` represents the root directory of the DOS disk):

`mcopy -i {{path/to/image.img}} {{path/to/source_file}} ::/{{path/to/target_file}}`

- Copy a file from a disk image to the current directory:

`mcopy -i {{path/to/image.img}} ::/{{path/to/source_file}} .`

- Copy a directory recur[s]ively to an MS-DOS disk:

> `mcopy -s {{path/to/source_directory}} {{A:/path/to/target_directory}}`

Reviewer: ```suggestion
`mcopy -s {{path/to/source_directory}} {{A}}:/{{path/to/target_directory}}`
```

- Copy a file and [p]reserve attributes:

> `mcopy -p {{path/to/source_file}} {{A:}}`

Reviewer: ```suggestion
`mcopy -p {{path/to/source_file}} {{A}}:/{{path/to/target_file}}`
```

I forgot to put this one
